buildSrc/src/main/groovy/com/uber/okbuck/composer/android/AndroidLibraryRuleComposer.groovy

```diff
@@ -46,6 +49,22 @@ final class AndroidLibraryRuleComposer extends AndroidBuckRuleComposer {
             testTargets.add(":${test(target)}")
         }
 
+        List<String> extraKotlincArgs = []
+        if (target.hasKotlinAndroidExtensions) {
```

Comment:
I would rather this be coming from the target than the composer. This block should not access anything about the target. The composers should be target api agnostic which is why we pass in all arguments via constructor. This keeps it decoupled and be able to extend to other cases.

Can you create a method in the android target instead that will return a list of kotlinc arguments and we can add logic for various extensions/plugins there instead?